app/src/main/java/de/tum/in/tumcampusapp/component/ui/studyroom/StudyRoomsActivity.java

```diff
@@ -168,32 +150,61 @@ public void openLink(View view) {
         startActivity(findStudyRoomIntent);
     }
 
-    @Override
-    protected Void onLoadInBackground(Void... arg) {
-        StudyRoomGroupManager sm = new StudyRoomGroupManager(this);
-        try {
-            sm.downloadFromExternal();
-        } catch (JSONException e) {
-            Utils.log(e);
-            // No error handling here
-        }
-        return null;
+    private void loadStudyRooms() {
+        showLoadingStart();
+        TUMCabeClient
+                .getInstance(this)
+                .getStudyRoomGroups(new Callback<List<StudyRoomGroup>>() {
+                    @Override
+                    public void onResponse(@NonNull Call<List<StudyRoomGroup>> call,
+                                           @NonNull Response<List<StudyRoomGroup>> response) {
+                        List<StudyRoomGroup> groups = response.body();
+                        if (response.isSuccessful() && groups != null && !groups.isEmpty()) {
+                            onDownloadSuccessful(groups);
+                        } else if (response.isSuccessful()) {
```

Comment:
I feel like an early return for `showErrorLayout()` would simplify the logic here